### PR TITLE
Remove footer, expand layout and move functions to other elements

### PR DESCRIPTION
**Pull request by melanieretter**
Goal of this PR is to remove the footer and expand the layout in vertical direction.

The following elements which are currently placed in the footer has been removed:
- Button "Drop here to delete"
- Button "No/You have Actions"

Every action which is triggered by the user is now performed at once. Deletions and assignments has to be confirmed. 

Instead of the button for deletion, every entity in a table has its own delete icon for deletion. It is possible to mark several entities and delete them together.

Assignments are performed after the desired entity is dropped and the confirmation question is not rejected.

Tags and Types can be created, updated or deleted by opening the menu (cogwheel symbol in the table header). When updating or deleting a tag/type an extra column with the specific action button is shown. 
After selecting the desired tag/type the delete action has to be confirmed as well / a pop-up window for updating the tag/type is opened.

Furthermore some small code glitches were fixed as well.

Signed-off-by: Melanie Retter <melanie.retter@bosch-si.com>

### CODE REVIEW

**Comment by schabdo** [comment]
@hawkbit-bot verify please

**Comment by schabdo** [comment]
@hawkbit-bot review please

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/filterlayout/AbstractFilterHeader.java [hunk 1]
```diff
@@ -10,99 +10,163 @@
...
 public abstract class AbstractFilterHeader extends VerticalLayout {
 
-    private static final long serialVersionUID = -1388340600522323332L;
+    private static final long serialVersionUID = 1L;
 
-    protected SpPermissionChecker permChecker;
+    private final SpPermissionChecker permChecker;
 
-    protected transient EventBus.UIEventBus eventBus;
+    private transient EventBus.UIEventBus eventBus;
```
Comment (stefbehl) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/filterlayout/AbstractFilterHeader.java]: add 'final'

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/table/AbstractTable.java [hunk 2]
```diff
@@ -54,23 +61,38 @@
...
+    protected static final String MESSAGE_CONFIRM_ASSIGN_MULTIPLE_ENTITIES = "message.confirm.assign.multiple.entities";
+
     private static final long serialVersionUID = 1L;
 
     private static final float DEFAULT_COLUMN_NAME_MIN_SIZE = 0.8F;
 
     protected static final String ACTION_NOT_ALLOWED_MSG = "message.action.not.allowed";
 
-    protected transient EventBus.UIEventBus eventBus;
+    private transient EventBus.UIEventBus eventBus;
```
Comment (stefbehl) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/table/AbstractTable.java]: add 'final'

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/filterlayout/AbstractFilterButtons.java [hunk 3]
```diff
@@ -27,30 +34,37 @@
...
-    private static final long serialVersionUID = 7783305719009746375L;
+    private static final long serialVersionUID = 1L;
 
-    private static final String DEFAULT_GREEN = "rgb(44,151,32)";
+    protected static final String DEFAULT_GREEN = "rgb(44,151,32)";
 
     protected static final String FILTER_BUTTON_COLUMN = "filterButton";
 
-    protected transient EventBus.UIEventBus eventBus;
+    private transient EventBus.UIEventBus eventBus;
```
Comment (stefbehl) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/filterlayout/AbstractFilterButtons.java]: add 'final'

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/table/AbstractTable.java [hunk 4]
```diff
@@ -237,18 +270,18 @@ protected void onBaseEntityEvent(final BaseUIEntityEvent<E> event) {
      *            the table transferable
      * @return set of entities id which will deleted
      */
-    public Set<Long> getDeletedEntityByTransferable(final TableTransferable transferable) {
+    public Set<Long> getSelectedEntitiesByTransferable(final TableTransferable transferable) {
```
Comment (stefbehl) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/table/AbstractTable.java]: typo 'tranferableData' below

**Comment by stefbehl** [comment]
@hawkbit-bot verify please

**Comment by schabdo** [comment]
@hawkbit-bot verify please

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/details/ArtifactDetailsLayout.java [hunk 5]
```diff
@@ -267,28 +267,25 @@ public Button generateCell(final Table source, final Object itemId, final Object
...
+                i18n.getMessage(SPUIDefinitions.BUTTON_OK), i18n.getMessage(SPUIDefinitions.BUTTON_CANCEL), ok -> {
                     if (ok) {
                         artifactManagement.delete(id);
                         uINotification.displaySuccess(i18n.getMessage("message.artifact.deleted", fileName));
                         final Optional<SoftwareModule> softwareModule = findSelectedSoftwareModule();
                         if (softwareModule.isPresent()) {
-                            populateArtifactDetails(softwareModule.get().getId(),
-                                    HawkbitCommonUtil.getFormattedNameVersion(softwareModule.get().getName(),
-                                            softwareModule.get().getVersion()));
+                            populateArtifactDetails(softwareModule.get());
```
Comment (bogdan-bondar) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/details/ArtifactDetailsLayout.java]: use ifPresent with OrElse instead of isPresent

Comment (bogdan-bondar) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/details/ArtifactDetailsLayout.java]: check if ifPresent() or orElse() can be used instead

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/table/AbstractTable.java (hunk 4 above)
Comment (melanieretter) [reply hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/table/AbstractTable.java]: fixed

**Pull request merged**